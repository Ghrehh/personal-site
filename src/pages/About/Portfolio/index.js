// @flow
import React from 'react';

import SectionHeading from 'components/SectionHeading';
import Button from 'components/Button';
import ButtonContainer from 'components/ButtonContainer';
import styles from './styles.module.css';

const Portfolio = () => (
  <section className={styles.display}>
    <div className={styles.displayInner}>
      <SectionHeading className={styles.heading}>PORTFOLIO</SectionHeading>

      <p className={styles.body}>
        I open-source most of my personal projects on GitHub. Why not check them
        out?
      </p>

      <ButtonContainer>
        <Button>View Portfolio</Button>
        <Button className={styles.gitHubButton}>GitHub</Button>
      </ButtonContainer>
    </div>
  </section>
);

export default Portfolio;
