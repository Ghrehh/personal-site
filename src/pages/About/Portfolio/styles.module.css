.display {
  background: linear-gradient(to bottom right, var(--purple), var(--orange));
  box-shadow: var(--box-shadow);
  padding: 53px;
}

.displayInner {
  max-width: 500px;
  margin: auto;
}

.heading {
  color: var(--orange);
  text-align: left;
}

.body {
  color: white;
}

.gitHubButton {
  background-color: transparent;
  border: 2px solid white;
  border-radius: 4px;
  text-align: center;
  width: 145px;
  padding: 10px;
  transition: transform 0.5s, background-color 0.3s, color 0.3s;
}

.gitHubButton:hover {
  filter: none;
  background-color: white;
  color: var(--charcoal);
}

@media (max-width: 500px) {
  .heading {
    color: var(--orange);
    text-align: center;
  }

  .display {
    text-align: center;
  }
}
