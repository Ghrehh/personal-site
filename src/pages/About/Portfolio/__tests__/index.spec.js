import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Portfolio from '../';

describe('Portfolio', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};
    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(<Portfolio />);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
