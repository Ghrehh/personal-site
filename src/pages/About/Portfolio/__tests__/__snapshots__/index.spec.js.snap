// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio renders the component correctly 1`] = `
<section
  className="display"
>
  <div
    className="displayInner"
  >
    <SectionHeading
      className="heading"
    >
      PORTFOLIO
    </SectionHeading>
    <p
      className="body"
    >
      I open-source most of my personal projects on GitHub. Why not check them out?
    </p>
    <ButtonContainer>
      <Button
        className=""
      >
        View Portfolio
      </Button>
      <Button
        className="gitHubButton"
      >
        GitHub
      </Button>
    </ButtonContainer>
  </div>
</section>
`;
